### ISSUE TITLE
Support Twitter's API 1.1 , v1 goes away in March 2013

### ISSUE THREAD

**Issue by akrherz** [dustin/twitty-twister#15]
Looks like version 1 is depreciated in early March!

https://dev.twitter.com/blog/planning-for-api-v1-retirement

Not sure if this breaks any of the code or not.  Just investigating myself.  Thanks again for this library

**Comment by akrherz** [comment]
Yeah, this breaks things.  For example, statuses/update.xml is no longer supported.

**Comment by ralphm** [comment]
I looked into it today. Changing over the API version isn't that hard. However, all API requests now have to be authenticated, with OAuth. Most of the examples have to be redone for this. And indeed XML is now no longer supported. Fortunately, the streaming API implementation is already parsing JSON, so I think we can change around some things so that those object classes can be used for the 'normal' API as well.

**Comment by akrherz** [comment]
Attempting to research this some more and am hitting this issue

https://dev.twitter.com/discussions/11280

The symptom was for statuses/update.json requests that worked in version 1 failed in 1.1 with http error 401 and twitter error code 32.  Figured I would dump something here as perhaps it may help somebody else.  I'm still boggling how to fix it.

_update_ , looks like the problem is with messages with a '/' in them
`twitter.update( 'test test')` works
`twitter.update( 'test / test')` fails

urllib.quote does not convert / , maybe it should be urllib.quote_plus ?

**Comment by akrherz** [comment]
7 May 2013 is the last day for v1
https://dev.twitter.com/blog/api-v1-retirement-final-dates

**Comment by ralphm** [comment]
As the retirement date for the v1 API has been extended to June 11, I took some time to look at this again.
- I moved the abstraction objects to their own module `twittytwisted.platform`.
- As all requests must be authenticated now, using OAuth, I removed all non-OAuth authentication code, raising a warning when you try to use the old `user`/`passwd` and raising an exception when you didn't provide OAuth credentials. The examples still need to be adjusted.
- I started using [Treq](http://github.com/dreid/treq) for rewriting the request code. This is a nice request API on top of the newer Twisted Web Agents code, and allows for removing a bulk of custom request code currently in twitty-twister. Also, it supports HTTP 1.1, unlike the existing code that still uses HTTP 1.0, which is [now deprecated](https://dev.twitter.com/blog/deprecating-http-1.0-streaming-api).

It turns out that a bunch of the methods use API endpoints that were either deprecated or removed in API v1, and no longer exist in v1.1. Also, the way paging is done has changed and the old way doesn't work anymore. I'm wondering if it is worthwhile to convert these methods or just drop them.

To whoever is reading this, which methods of `twittytwisted.twitter.Twitter` do you use? Also, what do you think of the current way it works, with delegate functions that are called for each item for API calls that return a list of things?

**Comment by mmichie** [comment]
Hey just started playing with this library yesterday and just now realized the update method doesn't work with twittytwisted.twitter.Twitter.  I can get streaming updates with the TwitterFeed, but I need to be able to write out status to the timeline.

What can I do to help?

**Comment by ralphm** [comment]
@mmichie: basically Twitter deprecated and shutdown the old API endpoints. This library supports v1.1 for the Streaming API, but the other APIs, like the update one you mention, require a whole bunch of non-trivial changes as outlined above. So far, nobody has expressed enough interest to work on this, and I personally only use the Streaming API stuff.
